workspace/build_manifest.go

```diff
@@ -59,41 +58,53 @@ type ExecPhase struct {
 
 func (e *ExecPhase) EnvironmentVariables(ctx context.Context, envName string, data runtime.RuntimeEnvironmentData) []string {
 
-	result := make([]string, 0)
+	envMap := make(map[string]string)
 
 	for _, property := range e.Environment["default"] {
 
 		if _, _, ok := plumbing.SaneEnvironmentVar(property); ok {
 			interpolated, err := TemplateToString(property, data)
 			if err == nil {
-				result = append(result, interpolated)
-			} else {
-				result = append(result, property)
+				if key, value, ok := plumbing.SaneEnvironmentVar(interpolated); ok {
```

Comment:
The double usage of `SaneEnvironmentVar` is confusing here. Part of the difficulty is its name: it's not evocative of the fact that it splits the string. I recommend renaming the function and either add comments to describe the purpose of this loop, or rework to have less conditionals.

Reply:
Will do